src/FX.Tests/E2ETests.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.FSharp.Core;
+
+using NUnit.Framework;
+using StackExchange.Redis;
+
+using FsharpExchangeDotNetStandard;
+using GrpcModels;
+
+namespace FsharpExchange.Tests
+{
+    [TestFixture]
+    public class E2ETests
+    {
+        private Process LaunchGrpcServer()
+        {
+            var solutionDir = (new DirectoryInfo(Environment.CurrentDirectory)).Parent?.Parent?.Parent?.Parent?.Parent;
+            var serviceExeDir = Path.Join(solutionDir.FullName, "src", "FX.GrpcService", "bin", "Debug", "net8.0");
+
+            var argsString = $"--urls https://localhost:{GrpcClient.Instance.HttpsPort}";
+
+            if (OperatingSystem.IsWindows())
+            {
+                var serviceExePath = Path.Join(serviceExeDir, "FX.GrpcService.exe");
+                return Process.Start(serviceExePath, argsString);
+            }
+            else 
+            {
+                var processInfo = new ProcessStartInfo
+                {
+                    UseShellExecute = false,
+                    CreateNoWindow = true,
+                    FileName = "dotnet",
+                    Arguments = Path.Join(serviceExeDir, "FX.GrpcService.dll") + " " + argsString
+                };
+                return Process.Start(processInfo);
+            }
+        }
+
+        [Test]
+        async public Task GrpcE2ETest()
+        {
+            using var serverProcess = LaunchGrpcServer();
+            await Task.Delay(TimeSpan.FromSeconds(1.0));
+
+            var client = new GrpcClient.Instance();
+            client.Connect();
+            
+            var order = new GrpcModels.LimitOrder(1.0m, Side.Ask.ToString(), 1.0m);
+            var response = await client.SendMessage<GrpcModels.LimitOrder, FSharpOption<Match>>(order);
+
+            Assert.That(response.Value.IsFull);
```

Comment:
@webwarrior-ws what does IsFull mean?

Reply:
Auto-generated property on DU checking if value is of case `Full`

Reply:
what? I don't understand; the way we were aiming to design this is to only have 2 types autogenerated: GenericInputParam and GenericOutputParam (they are the only ones that exist in the .proto file actually).

Reply:
I think it is auto-generated by F# compiler. Since I don't see any properties on `Match` type, but auto-complete suggests `IsFull` and `IsPartial` properties, and code does compile.

Reply:
Ah right, I guess that's the way for dotnet to expose F# DUs to C#. Marking this convo as resolved then (but don't forget following up in the other one).